cas-compute/src/symbolic/expr/mod.rs

```diff
@@ -964,4 +1026,70 @@ mod tests {
         let expr = parse_expr("(((((((((a) b) c) d) e + f) g) h) i) j)");
         assert_eq!(expr.to_string(), "j * i * h * g * (f + e * d * c * b * a)");
     }
+
+    #[test]
+    fn add_mul_commutative_hashing() {
+        // Commutative (should pass)
+        let eq_expressions: Vec<(&str, &str)> = vec![
+            // Addition
+            ("1 + 2", "2 + 1"),
+            ("x + y", "y + x"),
+            ("1 + x", "x + 1"),
+            ("a + b + c", "c + b + a"),
+            ("(x + y) + z", "z + (y + x)"),
+
+            // Multiplication (Explicit)
+            ("1 * 2", "2 * 1"),
+            ("x * y", "y * x"),
+            ("2 * x * y", "y * x * 2"),
+            ("a * b * c", "c * b * a"),
+            ("(x * y) * z", "z * (y * x)"),
+            ("x * (y + z)", "(y + z) * x"),
+            ("a * (b + c)", "(b + c) * a"),
+            ("1 + (x * y)", "(y * x) + 1"),
+
+            // Multiplication (Implicit)
+            ("1 2", "2 1"),
+            ("x y", "y x"),
+            ("2 x y", "y x 2"),
+            ("a b c", "c b a"),
+            ("(x y) z", "z * (y x)"),
+            ("x * (y + z)", "(y + z) x"),
+            ("a * (b + c)", "(b + c) a"),
```

Comment:
```suggestion
            ("(x y) z", "z (y x)"),
            ("x (y + z)", "(y + z) x"),
            ("a (b + c)", "(b + c) a"),
```

Most likely a typo :)

Reply:
Well, yeah it should be as you suggested. However, the issue is that `x (...)` is being interpreted as a function call `x(...)` rather than implicit multiplication. I'm not sure if this behavior is expected or not.

Reply:
I am starting a issue for this.

Reply:
I did not realize this was possible. Great catch!

Reply:
> I did not realize this was possible. Great catch!

Ah, so this indeed is unintended.